VKApp/VKApp/Custom/CustomInteractiveTransition.swift

```diff
@@ -0,0 +1,57 @@
+// CustomInteractiveTransition.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// Кастомизация жеста
+final class CustomInteractiveTransition: UIPercentDrivenInteractiveTransition {
+    // MARK: - Public Property
+
+    var isStarted = false
+    var viewController: UIViewController? {
+        didSet {
+            let recognizer = UIScreenEdgePanGestureRecognizer(
+                target: self,
+                action: #selector(handleScreenEdgeGestureAction(_:))
+            )
+            recognizer.edges = [.left]
+            viewController?.view.addGestureRecognizer(recognizer)
+        }
+    }
+
+    // MARK: - Private Property
+
+    private var isFinish = false
+
+    // MARK: - Private Methods
+
+    @objc private func handleScreenEdgeGestureAction(_ recognizer: UIScreenEdgePanGestureRecognizer) {
+        switch recognizer.state {
+        case .began:
+            isStarted = true
+            viewController?.navigationController?.popViewController(animated: true)
+        case .changed:
+            let translation = recognizer.translation(in: recognizer.view)
+            let relativeTranslation = translation.x / (recognizer.view?.bounds.width ?? 1)
+            let progress = max(0, min(1, relativeTranslation))
+            isFinish = progress > 0.33
```

Comment:
```suggestion
            let relativeTranslation = translation.x / (recognizer.view?.bounds.width ?? )
            let progress = max(, min(, relativeTranslation))
            isFinish = progress > 
```
цифры и коэффициенты выносим в константы

читаем статью про "Магические числа"